.github/workflows/test-rock.yaml

```diff
@@ -0,0 +1,72 @@
+name: Integration tests
+
+on:
+  workflow_call:
+    inputs:
+      rock-dir:
+        description: "Path to rock directory, i.e. directory containing rockcraft.yaml"
+        required: true
+        type: string
+      path-to-rock-file:
+        description: "Path to .rock file produced"
+        required: true
+        type: string
+      microk8s-channel:
+        description: "Microk8s channel e.g. 1.26/stable"
+        required: true
+        default: "1.26/stable"
+        type: string
+      juju-channel:
+        description: "Juju channel e.g. 3.1/stable"
+        required: true
+        default: "3.1/stable"
+        type: string
+      source-branch:
+        description: Github branch to checkout. If blank, it will use default values.
+        default: '' # For default behaviour, see https://github.com/actions/checkout/tree/v4/#usage
+        required: false
+        type: string
+jobs:
+  tests:
+    name: Test ${{ inputs.rock-dir }}
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.source-branch }}
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: ${{ inputs.rock-dir }}
+          path: ${{ inputs.rock-dir }}/
+
+      - name: Export ROCK to Docker
+        id: rock_in_docker
+        run: |
+          NAME=$(yq eval ".name" rockcraft.yaml)
+          VERSION=$(yq eval ".version" rockcraft.yaml)
+          DOCKER_IMAGE=$NAME:$VERSION
+          ROCK=$(echo ${{ inputs.path-to-rock-file }} | sed 's:.*/::')
+          sudo skopeo --insecure-policy copy oci-archive:$ROCK docker-daemon:$DOCKER_IMAGE
+          echo "image=$DOCKER_IMAGE" >> "$GITHUB_OUTPUT"
+        working-directory: ${{ inputs.rock-dir }}
+
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+            provider: microk8s
+            channel: ${{ inputs.microk8s-channel }}
+            charmcraft-channel: latest/candidate
+            microk8s-addons: "dns hostpath-storage rbac ingress metallb:10.64.140.43-10.64.140.49"
+            juju-channel: ${{ inputs.juju-channel }}
+
+      - name: Sanity tests
+        run: |
+          tox -e sanity
+        working-directory: ./${{ inputs.rock-dir }}
+
+      - name: Integration tests
+        run: |
+          tox -e integration
+        working-directory: ./${{ inputs.rock-dir }}
```

Comment:
If we do split these, I think I like it as separate files rather than jobs in a single `test-rock.yaml` just because the sanity tests are so much easier to implement than the integration tests, so I could see us having sanity tests everywhere and integration tests only where they're really needed

Reply:
Completely agreed as we discussed in our live session. Pushed a commit for this.